libasynql/src/poggit/libasynql/base/QueryRecvQueue.php

```diff
@@ -50,12 +50,10 @@ public function publishError(int $queryId, SqlError $error) : void{
 	}
 
 	public function fetchResults(&$queryId, &$results) : bool{
-		while($this->count() > 0){
-			$row = $this->shift();
-			if(is_string($row)){
-				[$queryId, $results] = unserialize($row, ["allowed_classes" => true]);
-				return true;
-			}
+		if($this->count() > 0){
```

Comment:
why can't you just change this to `while(($row = $this->shift()) !== null` here? Why do you need to check for the count?